batch_disposition.go

```diff
@@ -39,31 +44,35 @@ func (bdi *BatchDispositionIterator) Next() (uuid *uuid.UUID) {
 	return uuid
 }
 
-func (bdi *BatchDispositionIterator) doUpdate(ctx context.Context, ec entityConnector) error {
+func (bdi *BatchDispositionIterator) doUpdate(ctx context.Context, ec entityConnector) *BatchDispositionErrors {	
+	errors := map[*uuid.UUID] error {}
+
 	for !bdi.Done() {
 		if uuid := bdi.Next(); uuid != nil {
 			m := &Message{
 				LockToken: uuid,
 			}
 			m.ec = ec
 			err := m.sendDisposition(ctx, bdi.Status)
-			if err != nil {
-				return err
+			if err != nil {				
+				errors[uuid] = err
 			}
 		}
 	}
-	return nil
+	return &BatchDispositionErrors {
+		Errors: errors,
+	}
 }
 
 // SendBatchDisposition updates the LockToken id to the desired status.
-func (q *Queue) SendBatchDisposition(ctx context.Context, iterator BatchDispositionIterator) error {
+func (q *Queue) SendBatchDisposition(ctx context.Context, iterator BatchDispositionIterator) *BatchDispositionErrors {
```

Comment:
I'm teetering back and for on returning an error and specializing the error as described above. Thoughts?

Reply:
I agree, touch call. I do like the custom error type (see latest changeset) - but also seems odd. Maybe OnError, OnSuccess handlers?